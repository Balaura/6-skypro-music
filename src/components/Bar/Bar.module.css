.bar {
  position: fixed;
  bottom: 0;
  left: 0;
  width: 100%;
  background: rgba(28, 28, 28, 0.5);
}

.content {
  display: flex;
  flex-direction: column;
}

.playerBlock {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: 12px 20px;
}

@media (max-width: 768px) {
  .playerBlock {
    flex-wrap: wrap;
    justify-content: center;
  }

  .playerBlock > * {
    margin: 5px;
  }
}

.playerProgress {
  width: 100%;
  height: 5px;
  background: #2e2e2e;
}