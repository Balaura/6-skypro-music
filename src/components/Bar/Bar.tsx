"use client";

import React from 'react';
import Player from '@/components/Player/Player';
import TrackPlay from '@/components/TrackPlay/TrackPlay';
import Volume from '@/components/Volume/Volume';
import styles from './Bar.module.css';
import { ProgressBar } from '../ProgressBar/ProgressBar';
import { useSelector } from 'react-redux';
import { RootState } from '@/store/store';
import TimeTrack from '../TimeTrack/TimeTrack';
import Skeleton from '../Skeleton/Skeleton';

const Bar: React.FC = () => {
  const currentTrack = useSelector((state: RootState) => state.audioPlayer.currentTrack);
  const isLoading = useSelector((state: RootState) => state.audioPlayer.isLoading);

  if (!currentTrack) {
    return null; // Скрываем Bar, если нет текущего трека
  }

  {currentTrack && <TrackPlay currentTrack={currentTrack} />}  // для теста
  
    return (
      <div className={styles.bar}>
        <div className={styles.content}>
          <ProgressBar />
          <div className={styles.playerBlock}>
            <div className={styles.barPlayer}>
              <Player />
              {isLoading ? <Skeleton type="bar" /> : <TrackPlay currentTrack={currentTrack} /> }
                          </div>
            <TimeTrack />
            <Volume />
          </div>
        </div>
      </div>
    );
  }


export default Bar;