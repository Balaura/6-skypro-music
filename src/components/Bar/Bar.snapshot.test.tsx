import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Bar from './Bar';

const mockStore = configureStore([]);

// Мокируем все компоненты, которые использует Bar
jest.mock('@/components/Player/Player', () => 'Player');
jest.mock('@/components/TrackPlay/TrackPlay', () => 'TrackPlay');
jest.mock('@/components/Volume/Volume', () => 'Volume');
jest.mock('@/components/ProgressBar/ProgressBar', () => ({
  ProgressBar: () => 'ProgressBar'
}));
jest.mock('@/components/TimeTrack/TimeTrack', () => 'TimeTrack');
jest.mock('@/components/Skeleton/Skeleton', () => 'Skeleton');

describe('Bar component', () => {
  it('renders correctly', () => {
    const store = mockStore({
      audioPlayer: {
        currentTrack: {
          _id: 1,
          name: 'Test Track',
          author: 'Test Author',
        },
        isLoading: false,
      },
    });

    const tree = renderer.create(
      <Provider store={store}>
        <Bar />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});