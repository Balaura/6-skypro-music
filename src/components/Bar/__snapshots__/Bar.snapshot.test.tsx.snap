// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar component renders correctly 1`] = `
<div
  className="bar"
>
  <div
    className="content"
  >
    ProgressBar
    <div
      className="playerBlock"
    >
      <div
        className="barPlayer"
      >
        <Player />
        <TrackPlay
          currentTrack={
            {
              "_id": 1,
              "author": "Test Author",
              "name": "Test Track",
            }
          }
        />
      </div>
      <TimeTrack />
      <Volume />
    </div>
  </div>
</div>
`;
